### PR TITLE
Update Smoke CO Alarm Implementation

### PR DESCRIPTION
**Pull request by ericzijian1994**
Main updates are as follows:
1. Add `GetExpiryDate` function to accommodate cluster changes.
2. Add the conflict handling of DeviceMuted and Critical alarms.
3. Add `emberAfHandleEventTrigger` function and build description for manufacturer testing.

### CODE REVIEW

**Review by robszewczyk — DISMISSED** [review]

File: examples/smoke-co-alarm-app/silabs/src/ZclCallbacks.cpp [hunk 1]
```diff
@@ -65,3 +66,184 @@ bool emberAfPluginSmokeCoAlarmSelfTestRequestCommand(EndpointId endpointId)
...
+        if (success)
+        {
+            success = SmokeCoAlarmServer::Instance().SetExpressedState(
+                1, chip::app::Clusters::SmokeCoAlarm::ExpressedStateEnum::kCOAlarm);
+        }
+        break;
+
+    case 0xffffffff000000a1:
+        success = SmokeCoAlarmServer::Instance().SetCOState(1, chip::app::Clusters::SmokeCoAlarm::AlarmStateEnum::kNormal);
+        if (success)
```
Comment (robszewczyk) [inline examples/smoke-co-alarm-app/silabs/src/ZclCallbacks.cpp:105]: Do you need checks on priority of expressed state?

File: examples/smoke-co-alarm-app/silabs/src/ZclCallbacks.cpp [hunk 2]
```diff
@@ -65,3 +66,184 @@ bool emberAfPluginSmokeCoAlarmSelfTestRequestCommand(EndpointId endpointId)
...
+            success =
+                SmokeCoAlarmServer::Instance().SetExpressedState(1, chip::app::Clusters::SmokeCoAlarm::ExpressedStateEnum::kNormal);
+        }
+        break;
+
+    case 0xffffffff00000095:
+        success = SmokeCoAlarmServer::Instance().SetBatteryAlert(1, chip::app::Clusters::SmokeCoAlarm::AlarmStateEnum::kWarning);
+        if (success)
+        {
+            success = SmokeCoAlarmServer::Instance().SetExpressedState(
```
Comment (robszewczyk) [inline examples/smoke-co-alarm-app/silabs/src/ZclCallbacks.cpp:116]: Do you need checks on priority of expressed state?

File: examples/smoke-co-alarm-app/silabs/src/ZclCallbacks.cpp [hunk 3]
```diff
@@ -65,3 +66,184 @@ bool emberAfPluginSmokeCoAlarmSelfTestRequestCommand(EndpointId endpointId)
...
+                1, chip::app::Clusters::SmokeCoAlarm::ExpressedStateEnum::kBatteryAlert);
+        }
+        break;
+
+    case 0xffffffff000000a5:
+        success = SmokeCoAlarmServer::Instance().SetBatteryAlert(1, chip::app::Clusters::SmokeCoAlarm::AlarmStateEnum::kNormal);
+        if (success)
+        {
+            success =
+                SmokeCoAlarmServer::Instance().SetExpressedState(1, chip::app::Clusters::SmokeCoAlarm::ExpressedStateEnum::kNormal);
```
Comment (robszewczyk) [inline examples/smoke-co-alarm-app/silabs/src/ZclCallbacks.cpp:126]: Do you need to check on what the expressed state should be after you've removed the battery alert?

File: examples/smoke-co-alarm-app/silabs/src/ZclCallbacks.cpp (hunk 1 above)
Comment (robszewczyk) [reply examples/smoke-co-alarm-app/silabs/src/ZclCallbacks.cpp:105]: Recommend creating a method that captures *some* priority on the underlying alarms and sets the `ExpressedState` attribute based on the underlying ongoing alarms

**Comment by ericzijian1994** [comment]
4. Refactor the underlying code of ExpressedState.
5. Add timer to simulate self-test.
6. Add manual self-test.

**Review by bzbarsky-apple — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Pull request merged**